src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -152,7 +154,8 @@ public function onKernelResponse(ResponseEvent $event)
             $request = $event->getRequest();
             $requestSessionCookieId = $request->cookies->get($sessionName);
 
-            if ($requestSessionCookieId && $session->isEmpty()) {
+            $isSessionEmpty = $session->isEmpty();
+            if ($requestSessionCookieId && $isSessionEmpty && $session->getUsageIndex() > 0) {
```

Comment:
why the check on usageIndex?

Reply:
To know if a session was destroyed by example [`invalidate`](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/Session/Session.php#L179) or [`migrate`](https://github.com/symfony/symfony/blob/4274a1f04707948dd0826ee37e1821154f4597af/src/Symfony/Component/HttpFoundation/Session/Session.php#L187). 

There is currently no other identicator currently to know if a session was destroyed.

The other way would be we to extend the `Session` with a new method `isDestroyed` to know when we need to clear the session cookie.

Reply:
But as mention by @devnix this does not solve the problem for native session being started and written without a session being used in symfony.